h2o-algos/src/main/java/hex/glm/ComputationState.java

```diff
@@ -404,8 +417,46 @@ private static void fillSubRange(int N, int c, int [] ids, double [] src, double
 
   public double [] betaMultinomial(){return _beta;}
 
-  public double [] betaMultinomial(int c, double [] beta) {return extractSubRange(_activeData.fullN()+1,c,_activeDataMultinomial[c].activeCols(),beta);}
-
+  public double [] betaMultinomial(int c, double [] beta) {return extractSubRange(_activeData.fullN()+1,c,
+          _activeDataMultinomial[c].activeCols(),beta);}
+
+  public double[] changeBetaGradient(int c, double[] oldBetaCnd, double[] betaCnd) {
+    if (oldBetaCnd.length != betaCnd.length) {
+      double[] betaCndTemp = new double[oldBetaCnd.length]; // expand betaCnd/_state._ginfo._graident to full length
+      _activeDataMultinomial[c]._activeCols = activeDataMultinomial()._activeCols;
+      fillSubRange(oldBetaCnd.length, 0, _activeDataMultinomial[c].activeCols(), betaCnd, betaCndTemp); // only copy one class
+      double[] gradientTemp = new double[betaCndTemp.length];
+      double[] oldGradient = ginfo()._gradient;
+      ginfo()._gradient = changeArrayWActiveCols(ginfo()._gradient, true);
+      fillSubRange(oldBetaCnd.length, 0, _activeDataMultinomial[c].activeCols(), ginfoMultinomial(c)._gradient,
+              gradientTemp);
+      ginfoMultinomial(c)._gradient = gradientTemp;
+      ginfo()._gradient = oldGradient;
+      return betaCndTemp;
+    }
+    return betaCnd;
+  }
+  
+  public double[] changeArrayWActiveCols(double[] shortenArray, boolean expand) {
+    if (!expand || (shortenArray != null && !(shortenArray.length == _totalBetaLength))) {  // ginfo is supposed to have full length
+      int[] activeCols = new int[shortenArray.length];
+      int offset = 0;
+      for (int classIndex = 0; classIndex < _nclasses; classIndex++) {
+        int[] activeColsOneClass = _activeDataMultinomial[classIndex].activeCols();
```

Comment:
`_activeDataMultinomial` seems not to be initialized. Some tests such as `tests/testdir_jira/runit_pubdev_1042_LBFGS_with_alpha_and_lambda_search.R` fail here with:

```
java.lang.NullPointerException
        at hex.glm.ComputationState.changeArrayWActiveCols(ComputationState.java:445)
        at hex.glm.ComputationState.updateState(ComputationState.java:787)
        at hex.glm.GLM$GLMDriver.fitLBFGS(GLM.java:1811)
        at hex.glm.GLM$GLMDriver.fitModel(GLM.java:2000)
        at hex.glm.GLM$GLMDriver.computeSubmodel(GLM.java:2445)
        at hex.glm.GLM$GLMDriver.doCompute(GLM.java:2581)
        at hex.glm.GLM$GLMDriver.computeImpl(GLM.java:2479)
        at hex.ModelBuilder$Driver.compute2(ModelBuilder.java:246)
        at hex.glm.GLM$GLMDriver.compute2(GLM.java:1156)
        at water.H2O$H2OCountedCompleter.compute(H2O.java:1637)
        at jsr166y.CountedCompleter.exec(CountedCompleter.java:468)
        at jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263)
        at jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:974)
        at jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1477)
        at jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)
```

Reply:
Yes, good catch!  Fixed.  Thank you.